clients/src/main/java/org/apache/kafka/clients/consumer/RetriableCommitFailedException.java

```diff
@@ -22,8 +22,22 @@
 
     private static final long serialVersionUID = 1L;
 
+    public static RetriableCommitFailedException withUnderlyingMessage(String additionalMessage) {
```

Comment:
I'm ok removing this API in spite of the compatibility concern. It's just that the other constructors are "standard" exception constructors and we have no real need to remove them.